### ISSUE TITLE
Simplify specifying custom ids and linking to numerics / writing numeric headers

### ISSUE THREAD

**Issue by DanielOaks** [ircdocs/modern-irc#122]
Markdown doesn't make it easy to specify custom IDs, especially for e.g. headers where being able to choose your own IDs would be good. With the current system it's also pretty painful to link to numerics (writing out stuff like `[ERR_NEEDMOREPARAMS] (#errneedmoreparams-461) (461)` with the correct quoting/numerics/link/etc is difficult).

Here are two examples of equivalent sections written in Markdown and in ReST:

```markdown
### VERSION message

         Command: VERSION
      Parameters: [<target>]

The `VERSION` command is used to query the version of the software and the [`RPL_ISUPPORT`](#rplisupport-parameters) parameters of the given server. If `<target>` is not given, the information for the server the client is connected to should be returned.

If `<target>` is a server, the information for that server is requested. If `<target>` is a client, the information for the server that client is connected to is requested. If `<target>` is given and a matching server cannot be found, the server will respond with the `ERR_NOSUCHSERVER` numeric and the command will fail.

Wildcards are allowed in the `<target>` parameter.

Upon receiving a `VERSION` command, the given server SHOULD respond with one `RPL_VERSION` reply and one or more `RPL_ISUPPORT` replies.

Numeric Replies:

* [`ERR_NOSUCHSERVER`](#errnosuchserver-402) `(402)`
* [`RPL_ISUPPORT`](#rplisupport-005) `(005)`
* [`RPL_VERSION`](#rplversion-351) `(351)`

Command Examples:

      :Wiz VERSION *.se               ; message from Wiz to check the
                                      version of a server matching "*.se"

      VERSION tolsun.oulu.fi          ; check the version of server
                                      "tolsun.oulu.fi".
```

```rst
.. message-VERSION:

VERSION message
---------------

.. code-block:

         Command: VERSION
      Parameters: [<target>]

The ``VERSION`` command is used to query the version of the software and the :ref:`RPL_ISUPPORT <numeric-RPL_ISUPPORT-parameters>` parameters of the given server. If ``<target>`` is not given, the information for the server the client is connected to should be returned.

If ``<target>`` is a server, the information for that server is requested. If ``<target>`` is a client, the information for the server that client is connected to is requested. If ``<target>`` is given and a matching server cannot be found, the server will respond with the :numeric:`ERR_NOSUCHSERVER` numeric and the command will fail.

Wildcards are allowed in the ``<target>`` parameter.

Upon receiving a ``VERSION`` command, the given server SHOULD respond with one ``RPL_VERSION`` reply and one or more ``RPL_ISUPPORT`` replies.

Numeric Replies:

* :numeric:`ERR_NOSUCHSERVER`
* :numeric:`RPL_ISUPPORT`
* :numeric:`RPL_VERSION`

Command Examples::

      :Wiz VERSION *.se               ; message from Wiz to check the
                                      version of a server matching "*.se"

      VERSION tolsun.oulu.fi          ; check the version of server
                                      "tolsun.oulu.fi".
```

-----

I really want to keep this site using Jekyll and Markdown, because I like both of those and because all of the ircdocs sites use those.

But yeah, we do need to have a simpler way of specifying IDs and of linking to numerics especially. In this issue I'll lay out what I wanna do.

**Comment by DanielOaks** [comment]
Writing up a couple of custom plugins now, this is how it could look. I could probably also create a convenience tag to create the message header (will do the same to create numeric headers).

```md
{% h3 message-VERSION %}VERSION message{% endh3 %}

         Command: VERSION
      Parameters: [<target>]

The `VERSION` command is used to query the version of the software and the [`RPL_ISUPPORT` parameters](#RPL_ISUPPORT-parameters) of the given server. If `<target>` is not given, the information for the server the client is connected to should be returned.

If `<target>` is a server, the information for that server is requested. If `<target>` is a client, the information for the server that client is connected to is requested. If `<target>` is given and a matching server cannot be found, the server will respond with the {% numeric ERR_NOSUCHSERVER %} numeric and the command will fail.

Wildcards are allowed in the `<target>` parameter.

Upon receiving a `VERSION` command, the given server SHOULD respond with one `RPL_VERSION` reply and one or more `RPL_ISUPPORT` replies.

Numeric Replies:

* {% numeric ERR_NOSUCHSERVER %}
* {% numeric RPL_ISUPPORT %}
* {% numeric RPL_VERSION %}

Command Examples:

      :Wiz VERSION *.se               ; message from Wiz to check the
                                      version of a server matching "*.se"

      VERSION tolsun.oulu.fi          ; check the version of server
                                      "tolsun.oulu.fi".
```

**Comment by DanielOaks** [comment]
liquid tags:

- `{% numericheader RPL_ISUPPORT %}` : Header for the `RPL_ISUPPORT` numeric.
- `{% numeric RPL_ISUPPORT %}` : Link to the `RPL_ISUPPORT` numeric, including the name and the numeric.
- `{% messageheader VERSION %}` : Header for the `VERSION` message/command.
- `{% message VERSION %}` : Link to the `VERSION` message/command.

The numeric ones will use a data file to get the numeric itself and etc. We'll look to display the links the same way they're displayed currently.

**Comment by DanielOaks** [comment]
I'll consider this closed with #123, that PR is ready to be merged and adds the above tags (as well as the `{% command %}` tag to also link to messages, and the `{% isupport %}` tag to link to ISUPPORT params). I think we can continue discussion over in that PR \o/
